integration_tests/src/main/python/cast_test.py

```diff
@@ -151,6 +151,7 @@ def test_cast_string_date_non_ansi():
                                       StringGen('[0-9]{1,4}-[0-3][0-9]-[0-5][0-9][ |T][0-3][0-9]:[0-6][0-9]:[0-6][0-9]'),
                                       StringGen('[0-9]{1,4}-[0-3][0-9]-[0-5][0-9][ |T][0-3][0-9]:[0-6][0-9]:[0-6][0-9].[0-9]{0,6}Z?')],
                         ids=idfn)
```

Comment:
FYI. ```test_cast_string_date_valid_ansi``` and ```test_cast_string_date_valid_format``` need for xfail as well. It's skipped before Spark 320.

Reply:
This should be an existing bug.

Reply:
Yes, per offline discussed. Previous type check missed some cases about date data type which is also timezone aware.